pkg/asset/machines/aws/aws.go

```diff
@@ -0,0 +1,93 @@
+// Package aws generates Machine objects for aws.
+package aws
+
+import (
+	"text/template"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Config is used to generate the machine.
+type Config struct {
+	ClusterName string
+	Replicas    int64
+	Tags        map[string]string
+	Region      string
+	Machine     types.AWSMachinePoolPlatform
+}
+
+// WorkerMachineSetTmpl is template for worker machineset.
+var WorkerMachineSetTmpl = template.Must(template.New("aws-worker-machineset").Parse(`
+apiVersion: cluster.k8s.io/v1alpha1
+kind: MachineSet
+metadata:
+  name: {{.ClusterName}}-worker-0
+  namespace: openshift-cluster-api
+  labels:
+    sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
+    sigs.k8s.io/cluster-api-machine-role: worker
+    sigs.k8s.io/cluster-api-machine-type: worker
+spec:
+  replicas: {{.Replicas}}
+  selector:
+    matchLabels:
+      sigs.k8s.io/cluster-api-machineset: worker
+      sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
+  template:
+    metadata:
+      labels:
+        sigs.k8s.io/cluster-api-machineset: worker
+        sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
+        sigs.k8s.io/cluster-api-machine-role: worker
+        sigs.k8s.io/cluster-api-machine-type: worker
+    spec:
+      providerConfig:
+        value:
+          apiVersion: aws.cluster.k8s.io/v1alpha1
+          kind: AWSMachineProviderConfig
+          ami:
+            filters:
+              - name: "name"
+                values:
+                - rhcos*
```

Comment:
We don't want all of this filtering, do we?  I think we want to calculate the AMI ID in Go and inject it directly, like we're currently doing [over here][1].  That will also make it easier to allow callers to override the AMI ID if/when we restore support for that.

[1]: https://github.com/openshift/installer/blob/v0.2.0/pkg/asset/manifests/machine-api-operator.go#L100-L103

Reply:
> `InstallConfig` does not have AMI override. So we don't support that for workers.

But we do support AMI lookup in Go.  And we use that lookup to set the master/bootstrap AMI ID [here][1].  I think we want to use the same AMI for workers, which means we should pull that lookup out to a higher level so we don't have to perform it twice.  `InstallConfig` seems like a reasonable place to put it, but I'm ok if you want to stick it somewhere else for the time being.  I don't think we need to supply a user prompt for it.

[1]: https://github.com/openshift/installer/blob/v0.2.0/pkg/tfvars/tfvars.go#L89-L103

Reply:
> > I think we want to use the same AMI for workers.
>
> That's not true. New workers created by scaling the machineset should not be stuck with old amis.
> Bootstrap and master is separate because we create them.

And with this PR, we're effectively creating the workers as well.

> But in future we want masters to be adopted by the something similar

Bootstrap doesn't matter; since it's a throw-away node.  I expect we'll want a machine-set for masters too (or is there a different approach to scaling masters?).  And however we handle rolling updates for workers, I expect we'll want the same handling for masters.  Isn't that what the machine-config operator is going to handle?  Can't it update the machine-sets when it decides to bump to a new AMI?  For now, I thought we'd want to punt on all of this and just pin to the creation-time AMI for both masters and workers.

Reply:
>  Isn't that what the machine-config operator is going to handle? Can't it update the machine-sets when it decides to bump to a new AMI?

it doesn't control machine characteristics. So no.

>  I thought we'd want to punt on all of this and just pin to the creation-time AMI for both masters and workers.

I would still want installer to not choose the ami for worker.

Reply:
> Who is (or will be?) in charge of rolling AMI updates out across masters/workers?

That is TBD.

> But you're fine having it chose the AMI for the masters?

I don't want to do that for masters too. but we cannot do that now as we cannot run the cluster-api stack during bootstrap, because of various reasons, to give us masters.

Reply:
Switched to specific AMI to keep consistency with current MAO implementation.